core/fetcher/fetcher.go

```diff
@@ -292,3 +300,61 @@ func (f *Fetcher) fetchBuilderProposerData(ctx context.Context, slot int64, defS
 
 	return resp, nil
 }
+
+// fetchContributionData fetches the sync committee contribution data.
+func (f *Fetcher) fetchContributionData(ctx context.Context, slot int64, defSet core.DutyDefinitionSet) (core.UnsignedDataSet, error) {
+	resp := make(core.UnsignedDataSet)
+	for pubkey := range defSet {
+		// Query AggSigDB for DutyPrepareSyncContribution to get sync committee selection.
+		selectionData, err := f.aggSigDBFunc(ctx, core.NewPrepareSyncContributionDuty(slot), pubkey)
+		if err != nil {
+			return core.UnsignedDataSet{}, err
+		}
+
+		selection, ok := selectionData.(core.SyncCommitteeSelection)
+		if !ok {
+			return core.UnsignedDataSet{}, errors.New("invalid sync committee selection")
+		}
+
+		subcommIdx := uint64(selection.SubcommitteeIndex)
+
+		// Check if the validator is an aggregator for the sync committee.
+		ok, err = eth2exp.IsSyncCommAggregator(ctx, f.eth2Cl, selection.SelectionProof)
+		if err != nil {
+			return core.UnsignedDataSet{}, err
+		} else if !ok {
+			log.Debug(ctx, "Validator not selected for sync committee contribution duty", z.Any("pubkey", pubkey))
+			continue
+		}
+
+		// Query AggSigDB for DutySyncMessage to get beacon block root.
+		syncMsgData, err := f.aggSigDBFunc(ctx, core.NewSyncMessageDuty(slot), pubkey)
+		if err != nil {
+			return core.UnsignedDataSet{}, err
+		}
+
+		msg, ok := syncMsgData.(core.SignedSyncMessage)
+		if !ok {
+			return core.UnsignedDataSet{}, errors.New("invalid sync committee message")
+		}
+
+		blockRoot := msg.BeaconBlockRoot
+
+		// Query BN for sync committee contribution.
+		contribution, err := f.eth2Cl.SyncCommitteeContribution(ctx, eth2p0.Slot(slot), subcommIdx, blockRoot)
+		if err != nil {
+			return core.UnsignedDataSet{}, err
+		} else if contribution == nil {
+			// Some beacon nodes return nil if the beacon block root is not found for the subcommittee, return retryable error.
+			// This could happen if the beacon node didn't subscribe to the correct subnet.
+			return core.UnsignedDataSet{}, errors.New("sync committee contribution not found", z.U64("subcommIdx", subcommIdx), z.Hex("root", blockRoot[:]))
```

Comment:
your error is not retryable.... `sync committee contribution not found by root (retryable)`

Reply:
also, please use `snake_case` for structured logging fields